generate/generate_report_test.go

```diff
@@ -250,10 +251,33 @@ func assertReportHasError(t *testing.T, report generate.Report, err error) {
 
 func assertEqualReports(t *testing.T, got, want generate.Report) {
 	t.Helper()
-	if diff := cmp.Diff(got, want); diff != "" {
+
+	if diff := cmp.Diff(got, want, cmp.Comparer(compareErrors)); diff != "" {
 		t.Errorf("got %s", got)
 		t.Errorf("want %s", want)
 		t.Errorf("got(-) want(+)")
 		t.Fatal(diff)
 	}
 }
+
+func compareErrors(err, err2 error) bool {
```

Comment:
not nice to have this duplication just for the diff.
I think it'd be better to call cmp.Diff() on the individual fields of the generate.Report but call errtest.Assert() for the errors.

Reply:
I was in the fence between that and just providing a cmp function, it got more involved than what I initially thought it would be.

Reply:
I changed it to use the individual fields comparison, seems to be working well. The tradeoff is that for a complex structure like the report it is not extremely simple + as the report evolves it would be easy to end up not comparing the report properly, forgetting fields, etc. Just to ellaborate to why I was on the fence + for me it is not as clear that the field comparison will be better and the alternative was "not nice"...neither are very nice IMHO :laughing: